scipy/interpolate/tests/test_fitpack2.py

```diff
@@ -958,6 +1035,11 @@ def test_derivatives(self):
         assert_array_almost_equal(lut(x,y,dy=1,grid=False),dy)
         assert_array_almost_equal(lut(x,y,dx=1,dy=1,grid=False),dxdy)
 
+        lut_custom = regrid_python.regrid_python(x,y,z)
+        assert_array_almost_equal(lut_custom(x,y,dx=1,grid=False),dx)
+        assert_array_almost_equal(lut_custom(x,y,dy=1,grid=False),dy)
```

Comment:
`grid={True, False}` is ugly. Let's see what will it take to get rid of this. If we follow `RegularGridInterpolator` and accept an array of shape `(...., ndim)`, how much complexity this shifts onto the user?